### ISSUE TITLE
BUG: Pandas verbosely prints CSV contents

### ISSUE THREAD

**Issue by susops** [pandas-dev/pandas#56302]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
info = pd.read_csv(
            some_raw_csv_file_contents,
            escapechar='\n',
            skip_blank_lines=True,
            on_bad_lines='skip',
            header=None,
            nrows=some_number
)
return info
```

### Issue Description

When using the read_csv method when providing raw CSV contents, in this case about 115MB worth this method would persistently print all the contents to the screen. I used the breakpoint session and confirmed it was right when this method is called. 

### Expected Behavior

Regardless of whether passing raw csv contents to this method is efficient or not, the constant printing made it really difficult to troubleshoot and slowed things down. I am currently working around this by saving the contents temporarily using the tempfile module, and providing a file path instead of raw contents. I now don't get printing to the screen. Not sure if this is intended or a bug, but it was difficult to work with. 

### Installed Versions

>>> pandas.show_versions(as_json=False)
/usr/local/lib/python3.11/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")

INSTALLED VERSIONS
------------------
commit              : e86ed377639948c64c429059127bcf5b359ab6be
python              : 3.11.4.final.0
python-bits         : 64
OS                  : Linux
OS-release          : 5.15.133.1-microsoft-standard-WSL2
Version             : #1 SMP Thu Oct 5 21:02:42 UTC 2023
machine             : x86_64
processor           : 
byteorder           : little
LC_ALL              : None
LANG                : C.UTF-8
LOCALE              : en_US.UTF-8

pandas              : 2.1.1
numpy               : 1.24.3
pytz                : 2023.3.post1

**Comment by phofl** [comment]
Can you please provide a reproducer that we can copy and paste? Your input needs to be specified in some form

**Comment by susops** [comment]
Yes sorry, I guess I was expecting a message like "invalid file-path", which it does with a much smaller string. However, seems the really big string just complicated things. Thanks!
